fbgemm_gpu/codegen/embedding_backward_split_indice_weights_template.cu

```diff
@@ -245,6 +245,18 @@ Tensor {{ ddesc }}_embedding_codegen_grad_indice_weights{{ vdesc }}_cuda(
         TENSOR_ON_CUDA_GPU(feature_requires_grad);
     }
 
+    auto aligned_grad_output = grad_output.clone();
+    // FIXME: to support aligned memory access in Vec4T load/store function
+    // 16 for FP32 and 8 for FP16
+    if (grad_output.dim() > 1 &&
+        (reinterpret_cast<uint64_t>(grad_output.data_ptr()) % 16 != 0 ||
+        grad_output.stride(1) != 1 || grad_output.stride(0) % 4 != 0)) {
+        aligned_grad_output = grad_output.contiguous();
+    }
+    if (reinterpret_cast<uint64_t>(grad_output.data_ptr()) % 16 != 0) {
+        aligned_grad_output = at::empty_like(grad_output).copy_(grad_output);
+    }
```

Comment:
I guess the code got moved here. I guess you're only running this inside of the CUDA kernel now?

Reply:
One potential hazard to be aware of when doing a transform like this, is if the operator you moved this logic into is differentiable. The backward in that case may have been relying on the input being guaranteed to be aligned, including the saved copy for backwards.